ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountStatsMySqlStore.java

```diff
@@ -310,18 +322,179 @@ public String queryRecordedMonth(String clusterName) throws SQLException {
   }
 
+  @Override
+  public synchronized void storePartitionClassStats(StatsWrapper statsWrapper) throws SQLException {
+    long startTimeMs = System.currentTimeMillis();
+    // 1. Get all the partition class names in this statswrapper
+    Map<String, StatsSnapshot> partitionClassSubMap = statsWrapper.getSnapshot().getSubMap();
+    Set<String> partitionClassNames = new HashSet<>(partitionClassSubMap.keySet());
+    Map<String, Short> partitionClassNamesInDB = partitionClassReportsDao.queryPartitionClassNames(clusterName);
+    // 2. Add partition class names not in db
+    partitionClassNames.removeAll(partitionClassNamesInDB.keySet());
+    if (!partitionClassNames.isEmpty()) {
+      for (String partitionClassName : partitionClassNames) {
+        partitionClassReportsDao.insertPartitionClassName(clusterName, partitionClassName);
+      }
+      // Fresh the partition class names
+      partitionClassNamesInDB = partitionClassReportsDao.queryPartitionClassNames(clusterName);
+    }
+
+    // 3. Get partition ids under each partition class name
+    Map<String, List<Short>> partitionIdsUnderClassNames = partitionClassSubMap.entrySet()
+        .stream()
+        .collect(Collectors.toMap(Map.Entry::getKey, ent -> ent.getValue()
+            .getSubMap()
+            .keySet()
+            .stream()
+            .map(pk -> Short.valueOf(pk.substring("Partition[".length(), pk.length() - 1)))
+            .collect(Collectors.toList())));
+
+    // 4. Add partition ids not in db
+    Set<Short> partitionIdsInDB = partitionClassReportsDao.queryPartitionIds(clusterName);
+    for (String partitionClassName : partitionIdsUnderClassNames.keySet()) {
+      short partitionClassId = partitionClassNamesInDB.get(partitionClassName);
+      for (Short pid : partitionIdsUnderClassNames.get(partitionClassName)) {
+        if (!partitionIdsInDB.contains(pid)) {
+          partitionClassReportsDao.insertPartitionId(clusterName, pid, partitionClassId);
+        }
+      }
+    }
+    storeMetrics.storePartitionClassStatsTimeMs.update(System.currentTimeMillis() - startTimeMs);
+  }
+
+  @Override
+  public synchronized void storeAggregatedPartitionClassStats(StatsSnapshot statsSnapshot) throws SQLException {
+    long startTimeMs = System.currentTimeMillis();
+    PartitionClassReportsDao.StorageBatchUpdater batch =
+        partitionClassReportsDao.new StorageBatchUpdater(config.updateBatchSize);
+    for (Map.Entry<String, StatsSnapshot> partitionClassMapEntry : statsSnapshot.getSubMap().entrySet()) {
+      String partitionClassName = partitionClassMapEntry.getKey();
+      for (Map.Entry<String, StatsSnapshot> accountContainerMapEntry : partitionClassMapEntry.getValue()
+          .getSubMap()
+          .entrySet()) {
+        String accountContainer = accountContainerMapEntry.getKey();
+        long usage = accountContainerMapEntry.getValue().getValue();
+        String[] parts = accountContainer.split(Utils.ACCOUNT_CONTAINER_SEPARATOR);
```

Comment:
Minor: could you add a comment with an example accountContainer string, to make it easier to understand the parsing logic?  It would be even better if accountContainer construction and deconstruction were in one class.